core/src/executor/evaluate/stateless.rs

```diff
@@ -165,6 +164,13 @@ fn evaluate_function<'a>(
 
             f::rtrim(name(), expr, chars)
         }
+        Function::Rand(expr) => {
+            let expr = match expr {
+                Some(e) => Some(eval(e)?),
+                _ => None,
+            };
+            f::rand(expr)
```

Comment:
[nit-picking]
refactor issue

```suggestion
            f::rand(expr.as_ref().map(|expr| eval(expr)).transpose()?)
```